Added multi column sorting. Fully compatible with regular and backend…

suneab opened the pull request:
… sorting and pagination

Fixes #
#1694

## Proposed Changes

- Adds the posibility of sorting on multiple columns
- Compatible with both frontend and backend sorting
- Compatible with both frontend and backend pagination

jtommy commented:
@suneab, please check my review.
I think we should add a way to reset the sort of a single column without resetting all.

suneab commented:
@jtommy i just pushed a change to fix the .find IE11 issue
I also removed the lodash package and added my own custom sorting method. i added it to the helper class because .sort method was interfering with array.sort

How would you like the column sort resetting to work? 

the only idea i can come up with is
click column for sorting asc <- how it works now
click same coulmn for sorting desc <- how it works now
click same column to reset it <- new change (not implimented yet!)

but i don't like it.. 

would appreciate your input.

suneab commented:
btw i left lodash orderBy for my fake backend sorting in the example. hope thats ok.

jtommy commented:
@suneab I think it's a good example https://handsontable.com/docs/7.1.1/demo-multicolumn-sorting.html

suneab commented:
I don't like it.. 
We keep the sorting priority when clicking the columns multiple time while only changing sorting direction, they change the priority every click.
If we added the reset on third click we would change the priority. 
Is there any other way of achieving this?

suneab commented:
Hi @jtommy 
I would appreciate some feedback :)

jtommy commented:
@suneab I don't know to be honest. Btw you should add a way to support multiple sort on mobile too (for example a dropdown)

suneab commented:
@jtommy i just added mobile support (IE11 is supported as well). Please let me know whats missing for this feature to be approved.

suneab commented:
oh btw, i couldnt figure out how to add the custom sorting icon to inside of the select options so i just added the HTML arrows
i found this thread, but its outdated and can't find anything in the current documentation about it
https://github.com/buefy/buefy/issues/74

jtommy commented:
@suneab any news about reset sorting column ?

suneab commented:
@jtommy nope, still dont know how to fit it in there

jtommy commented:
We have to find a way to reset or a way to enable multi sort on each column

suneab commented:
@jtommy i added a way to reset a single column by clicking a button. also works for mobile.

jtommy commented:
@suneab Nice idea! I'm checking your code and `custom-key` prop is too generic name, please change it to `sort-multiple-key`

suneab commented:
i fixed the naming of the custom key. Not sure what would be the correct versions in the package file

jtommy commented:
@suneab i don't see your commit and please reset your package-lock.json

jtommy commented on src/components/table/Table.vue:
```diff
@@ -91,7 +94,29 @@
 
+                                <template
+                                    v-if="sortMultiple &&
+                                        sortMultipleDataComputed.find(i =>
```
Please don't use find method

jtommy commented on src/components/table/Table.vue:
```diff
@@ -91,7 +94,29 @@
+                                        :class="{ 'is-desc': sortMultipleDataComputed.find(i =>
+                                        i.field === column.field).order === 'desc'}"
+                                    />
+                                    {{ sortMultipleDataComputed.findIndex(i =>
```
Please don't use findIndex

jtommy commented on src/components/table/Table.vue:
```diff
@@ -91,7 +94,29 @@
+                                    {{ sortMultipleDataComputed.findIndex(i =>
+                                    i.field === column.field) + 1 }}
+                                    <button
+                                        style="margin-left:10px"
```
Please add a class that set margin and apply on it

suneab commented:
> @suneab i don't see your commit and please reset your package-lock.json

lol i forgot to press commit.. i fixed the package as well

jtommy commented:
@suneab please resolve conflicts so I'll merge it, thanks :)

suneab commented:
conflicts resolved

ms-saltation commented:
Thank you guys. Will try it out soon.
